barretenberg/cpp/src/barretenberg/vm/avm_trace/AvmMini_opcode.cpp

```diff
@@ -0,0 +1,153 @@
+#include "AvmMini_opcode.hpp"
+#include <cstdint>
+
+namespace avm_trace {
+
+const std::unordered_map<OpCode, size_t> Bytecode::OPERANDS_NUM = {
```

Comment:
Hmm I guess `OPERANDS_NUM` and `has_in_tag` could both turn into just fields of some class if you took a route similar to TS. Thoughts on that?

Reply:
If we take the path of defining an instruction type/class, I totally agree.